src/main/java/org/ru/skypro/coursework/AuctionSystem/service/ReportServiceImpl.java

```diff
@@ -0,0 +1,94 @@
+package org.ru.skypro.coursework.AuctionSystem.service;
+
+import org.ru.skypro.coursework.AuctionSystem.model.Report;
+import org.ru.skypro.coursework.AuctionSystem.model.projections.LotInfoForCSV;
+import org.ru.skypro.coursework.AuctionSystem.repository.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.io.ByteArrayResource;
+import org.springframework.core.io.Resource;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.time.LocalDateTime;
+import java.util.stream.Collectors;
+
+@Service
+public class ReportServiceImpl implements ReportService{
+    private static final Logger logger = LoggerFactory.getLogger(ReportServiceImpl.class);
+    private final LotRepository lotRepository;
+    private final ReportRepository reportRepository;
+
+    public ReportServiceImpl(LotRepository lotRepository, ReportRepository reportRepository) {
+        this.lotRepository = lotRepository;
+        this.reportRepository = reportRepository;
+    }
+
+    public String serializeLotInfoForCSV (LotInfoForCSV lotInfoForCSV) {
+        return lotInfoForCSV.getLotId()+
+                ";"+
+                lotInfoForCSV.getTitle()+
+                ";"+
+                lotInfoForCSV.getLotStatus()+
+                ";"+
+                lotInfoForCSV.getLastBidder()+
+                ";"+
+                lotInfoForCSV.getBidPrice()+
+                ";";
+   }
+
+    @Override
+    public ResponseEntity<Resource> getCSVFile() throws IOException {
+        String headers = "id;title;status;lastBidder;currentPrice;\n";
+        String csvTextReport = headers+
+                        lotRepository
+                        .findAllLotFullInfo()
+                        .stream()
+                        .map(LotInfoForCSV::fromLotFullInfo)
+                        .map(this::serializeLotInfoForCSV)
+                        .collect(Collectors.joining("\n"));
+
+        String fileName = "Lot_report_as_" + LocalDateTime.now() +
+                ".csv";
+        Path path = Path.of(
+                "src/main/java/org/ru/skypro/coursework/AuctionSystem",
+                "/REPORTS",
+                fileName);
+
+        Files.writeString(path, csvTextReport);
```

Comment:
Вот как реализовать через Apache Commons CSV.
Так же решается проблема с кодировкой.
```suggestion
        String fileName = "Lot_report_as_" + LocalDateTime.now() +
                ".csv";
        Path path = Path.of(
                "src/main/java/org/ru/skypro/coursework/AuctionSystem",
                "/REPORTS",
                fileName);

        try (Writer writer = new FileWriter(path.toFile());
             CSVPrinter csvPrinter = new CSVPrinter(writer, CSVFormat.DEFAULT)) {

            csvPrinter.printRecord("id", "title", "status", "startPrice", "bidPrice", "lastBidder");
            for (Object[] r : lotRepository.findAllForReport()) {
                csvPrinter.printRecord(r);
            }

            csvPrinter.flush();
            System.out.println("CSV file written successfully.");

        } catch (IOException e) {
            e.printStackTrace();
        }
```